### DISCUSSION TITLE
Agent config length limitation not works.

### DISCUSSION THREAD

**Discussion by weixiang1862** [apache/skywalking#11417] [General]
Agent config length limitation not works, the cutted propertyvalue should be returned by `cut` method in `ConfigInitializer.java`

![image](https://github.com/apache/skywalking/assets/16346644/c0980ccf-e2e3-4437-b38a-245b0099a689)

**Comment by wu-sheng** [comment]
Can't see your screenshot, please update what you want to show to us.

**Reply by weixiang1862** [reply]
Ok, I will describe it by text.

**Comment by weixiang1862** [comment]
As the annotation `org.apache.skywalking.apm.util.Length` description, it should has the ability to limit the length of a config propertyValue and `org.apache.skywalking.apm.util.ConfigInitializer` implement it in `initNextLevel` method by using `StringUtil.cut(propertyValue, lengthLimited)`.

But string is an immutable object, the changed value should be returned by cut method or it will keep as it before.
